### PR TITLE
nix: move ci dependencies to the devShell

### LINKED ISSUE #160: The default nix flake installable is heavy
**Is your feature request related to a problem? Please describe.**

Running `nix run github:numtide/treefmt` pulls about 5GB of dependencies.
It seems like the linters are somehow downloaded even when they are not used.

**Describe the solution you'd like**

The default app should be lightweight and downloading the linters tools should be optional.

### PR DESCRIPTION
**Pull request by TristanCacqueray**
Fixes #160

### CODE REVIEW

**Comment by zimbatm** [comment]
```
 = note: ld: library not found for -liconv
          clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```
I'm not sure why this is failing on macOS

**Comment by TristanCacqueray** [comment]
@zimbatm shouldn't the override appends to the treefmt buildInputs? e.g. to keep that definition: https://github.com/numtide/treefmt/blob/e7918e97ce19a5fbe91c0eb6bf336ed39f92f99d/default.nix#L49

**Comment by zimbatm** [comment]
doh. that's probably it :)

File: default.nix [hunk 1]
```diff
@@ -17,8 +17,23 @@ let
...
+  };
+
+  # Add all the dependencies of treefmt, plus more build tools
+  devShell = treefmt.overrideAttrs (prev: {
+    shellHook = ''
+      # Put the treefmt binary on the PATH when it's built
+      export PATH=$PWD/target/debug:$PATH
+    '';
+
+    buildInputs = prev.buildInputs ++ (with nixpkgs; [
```
Comment (TristanCacqueray) [inline default.nix:36]: Oops, I think that should be:

```suggestion
    nativeBuildInputs = (with nixpkgs; [
```

**Comment by zimbatm** [comment]
@TristanCacqueray nativeBuildInputs probably also needs to be concatenated with prev.nativeBuildInputs

**Comment by TristanCacqueray** [comment]
@zimbatm oh, ok I've squashed the change. We'll get through this eventually :)

**Comment by zimbatm** [comment]
success!

**Comment by TristanCacqueray** [comment]
nice, thanks!

**Pull request merged**